controllers/spec.go

```diff
@@ -148,7 +150,16 @@ func newBaseSpec(master *v1alpha1.CDAPMaster, name string, labels map[string]str
 	s.CConf = cconf
 	s.HConf = hconf
 	s.SysAppConf = sysappconf
-	s.ConfigMapVolumes = master.Spec.ConfigMapVolumes
+
+	s.ConfigMapVolumes = make(map[string]string)
+	for k, v := range master.Spec.ConfigMapVolumes {
+		s.ConfigMapVolumes[k] = v
+	}
```

Comment:
Agree that better to deep copy maps. Curious if shadow copy caused any sigsegv during your test, thus making this change?

Reply:
Mind helping with converting the label copy to deep copy as well where you are here. w

Reply:
Done.

I haven't tried shallow copy, but I think it is better to keep the `master` spec not mutated.